### ISSUE TITLE
[Feature Request] ol-cli should use abscissa_tokio

### ISSUE THREAD

**Issue by 0o-de-lally** [OLSF/libra#432]
TLDR; use abscissa_tokio for starting `warp` servers, instead of 'vanilla tokio`

`ol-cli` has the `serve` subcommand. This subcommand starts the server.rs main, which is a tokio async function. 

for abscissa apps, using tokio requires abscissa_tokio. Calling these async functions has a very specific implementation in abscissa_tokio.

In the current (vanilla tokio implementation) When calling some sync functions (from within a server.rs async block) will fail with:

```
Apr 05 18:02:57.153 ERROR reqwest::blocking::client: Failed to communicate successful startup: Ok(())    
The application panicked (crashed).
Message:  Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.
Location: /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/blocking/shutdown.rs:49
```

As server.rs is extended there may be other blocking functions that may be needed. So either server implements abscissa_tokio, or it can become a standalone app which does not use abscissa.

### Repro with:
A reproduction branch is here: `web-monitor-tokio-repro`

1. start the server `cd ol-cli/ && cargo r serve` (there will be no errrors, builds fine.
2. go to the webpage hosted in a browser http://localhost:3030 (you may need port forwarding)
3. navigate to http://localhost:3030/chain 
Expected result: should show a json feed updating every second.
Actual result: web browser shows blank page, and the process in terminal will show the above error. "cannot drop runtime"

### Notes on a potential abscissa_tokio implementation:

1. Similar apps using this method are: Sagan, and Delphi. They however use `warp 0.1`, which is different than `warp 0.2.5` which is used throughout libra.
- https://github.com/iqlusioninc/sagan
- https://github.com/iqlusioninc/delphi

2. In server_cmd.rs comments there is an example of how abscissa_tokio would be used: https://github.com/OLSF/libra/blob/3130456ed7c9bcb6168c5bdcbd09f829225be6e3/ol-cli/src/commands/serve_cmd.rs#L20-L25

3. Note that `trait` functions cannot await on async functions.

**Comment by lpgeiger** [comment]
@zmanian @tarcieri  am I describing the issue correctly?
